### ISSUE TITLE
Is it possible that data.y is also edges, like edge_index

### ISSUE THREAD

**Issue by zhangfuyang** [rusty1s/pytorch_geometric#660]
## ❓ Questions & Help

Hi. The thing I want to do is to predict the connection of the nodes. So the groundtruth label are the edges, specifically just like edge_index with shape[2, num_edges]. But the problem is that different graph has different number of edges in groundtruth, so it cannot train on min-batch. I know there is a solution that I can manually concatenate the groudtruth labels based on the Batch Attribute of DataLoader. But I wonder is there a more decent method to do that? e.g. like create a edge_index2 (same as edge_index) for replacing data.y.

Anyhow, this is really a great package that save me a lot of time. Thank you so much! really great work!

**Comment by rusty1s** [comment]
When you name your ground-truth edges `data.y_index` with shape `[2, num_edges]`, batching should work flawless for your idea. In fact, you can replace `data.y` with anything you like.

**Comment by zhangfuyang** [comment]
Thanks! `y_index` works. 
When you said `data.y` can be anything you mean it can be anything but only with same dimension for all graph in dataset right? So in this case, there will be an concatenation error when feed ground-truth directly into `data.y`.

**Comment by WMF1997** [comment]
hello @zhangfuyang:

In fact, ```Data``` is just a ```container```, you  can write anything you like... 


see the following screenshots. 

### ```data.y = ['cxk'] * 2333```
![image](https://user-images.githubusercontent.com/37621591/64084321-e5f16a80-cd5c-11e9-95f8-0c68c71d56da.png)

### what is in the graph. 
![image](https://user-images.githubusercontent.com/37621591/64084348-3e286c80-cd5d-11e9-881e-058e2006a444.png)

as shown in the 2 screenshots, it can really contains EVERYTHING YOU LIKE. 


yours sincerely
@wmf1997

**Comment by zhangfuyang** [comment]
Wow! Thank you so much WMF1997.

**Comment by evantianzg** [comment]
> hello @zhangfuyang:
> 
> ## Answer
> In fact, `Data` is just a `container`, you can write anything you like...
> 
> see the following screenshots.
> 
> ### `data.y = ['cxk'] * 2333`
> ![image](https://user-images.githubusercontent.com/37621591/64084321-e5f16a80-cd5c-11e9-95f8-0c68c71d56da.png)
> 
> ### What is in the graph.
> ![image](https://user-images.githubusercontent.com/37621591/64084348-3e286c80-cd5d-11e9-881e-058e2006a444.png)
> 
> as shown in the 2 screenshots, it can really contains EVERYTHING YOU LIKE.
> 
> ### P.S. Source code of `torch_geometric.data.Data`
> ```python
>    def __init__(self, x=None, edge_index=None, edge_attr=None, y=None,
>                  pos=None, norm=None, face=None, **kwargs):
>         self.x = x
>         self.edge_index = edge_index
>         self.edge_attr = edge_attr
>         self.y = y
>         self.pos = pos
>         self.norm = norm
>         self.face = face
>         for key, item in kwargs.items():
>             if key == 'num_nodes':
>                 self.__num_nodes__ = item
>             else:
>                 self[key] = item
> 
>         if torch_geometric.is_debug_enabled():
>             self.debug()
> ```
> 
> I only show the `__init__` method. In this method, it does not have the dimension check. `**kwargs` provides any extra attributes, for example, `train_idx`, which shows which node is to train. (this is seen in `Cora` dataset)
> 
> #### `data.cxk = 666`
> ![image](https://user-images.githubusercontent.com/37621591/64084481-7bd9c500-cd5e-11e9-9d0b-8200edaec4d5.png)
> 
> ### DO NOT MANAGE TO OVERWRITE `data.num_nodes`!
> sorry, `data.num_nodes` is an IMPORTANT ATTRIBUTE, as shown in the source code of `torch_geometric.data.Data`.
> 
> an example of overwriting it. ![image](https://user-images.githubusercontent.com/37621591/64084520-dd019880-cd5e-11e9-9126-8910add8031f.png)
> 
> ## Conclusion
> If you just want to try it FOR FUN, the code above shows some of the possible ways of using that. However, if you want to work with it. (especially in `debug` mode), CHECK IT CAREFULLY!
> 
> yours sincerely @WMF1997

Hi @WMF1997 
Really thanks for your reply. I am confused by the official description about data.y. Why the shape of data.y may be arbitrary? I understand the differences between node-level labels and graph-level labels. But really puzzled by the arbitrary shape.
